docs/developer/dependency_resolution.md

```diff
@@ -111,9 +111,13 @@ public class DefaultsExtension implements ServiceExtension {
 Provider methods configured with `isDefault=true` are only invoked, if the respective service (here: `FooStore`) is not
 provided by any other extension.
 
+> **WARNING**: Please do not abuse of `isDefault=true`! Especially we strongly discourage injecting services in default
```

Comment:
How about:
> "The `isDefault` attribute should be used sparingly, it typically is not needed for  technology extensions. Default provider methods are invoked during the _inject_-phase, i.e. while resolving `@Inject`-ed fields, so the code inside the provider method should not reference other fields, that are also `@Inject`-ed because they might not yet be initialized.
Their intended use is to provide default implementations for services, not to realize extensibility."